enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,500 @@
+
+### Goals
+
+* Explore the use of an operator for managing ExternalDNS.
+* Create patterns, libraries, and tooling so that ExternalDNS Operator is of high quality, consistent in its API
+surface (common fields on relevant ExternalDNS CRDs, consistent labeling of created resources, etc.), yet is easy to build.
+* Build an operator that is suitable for production use of ExternalDNS.
+* Focus on supporting the OpenShift Route resource as a source for ExternalDNS.
+* Additionally support the Kubernetes [service source](https://github.com/kubernetes-sigs/external-dns/blob/master/source/service.go) (namely LoadBalancer services)
+as sources for ExternalDNS, since OpenShift routes are not the only way to provide Ingress for a given workload.
+* Support ExternalDNS [providers](https://github.com/kubernetes-sigs/external-dns/tree/master/provider) relevant to OpenShift. This should at minimum
+include include AWS, Azure, and GCP in addition to BlueCat and Infoblox.
+
+### Non-Goals
+
+* Create an operator that is only compatible with OpenShift.
+* Replace the functionality of existing operators. __Note:__ The ExternalDNS Operator intends to eventually replace external DNS
+management provided by existing components, such as the [Ingress Operator](https://github.com/openshift/cluster-ingress-operator),
+in the long-term.
+* Support all available ExternalDNS [sources](https://github.com/kubernetes-sigs/external-dns/tree/master/source) and
+[providers](https://github.com/kubernetes-sigs/external-dns/tree/master/provider) (at least, initially).
+* Support the ExternalDNS [CRD source](https://github.com/kubernetes-sigs/external-dns/tree/master/docs/contributing/crd-source)
+by augmenting the existing OpenShift [ingressoperator/dnsrecord](https://github.com/openshift/api/blob/master/operatoringress/v1/types.go) resource.
+Support for the ExternalDNS CRD Source should be added to the ExternalDNS Operator in OCP 4.10.
+* Support Gateway APIs in ExternalDNS/ExternalDNS Operator. See [upstream issue 2045](https://github.com/kubernetes-sigs/external-dns/issues/2045)
+for more context. Support for Gateway APIs should be added after the initial "tech-preview" ExternalDNS Operator is available in OCP 4.9
+* Perform a gap analysis between ExternalDNS Operator and the [Ingress Operator's DNS provider implementations](https://github.com/openshift/cluster-ingress-operator/tree/master/pkg/dns).
+This could be performed as a part of the ExternalDNS Operator work for OCP 4.10.
+
+## Proposal
+
+The proposal is based on experience gathered and work accomplished with existing OpenShift operators.
+
+The following tasks would be completed as a part of this enhancement:
+
+* Create a GitHub repository to host the operator source code in OpenShift's GitHub org,
+with the hopes of moving the operator's source code into an upstream k8s-sigs repository in the future.
+* Leverage popular frameworks and libraries, e.g. controller-runtime and kubebuilder, to simplify development
```

Comment:
No OperatorSDK? It combines both kubebuilder and controller-runtime.

Reply:
Good question. We polled the openshift networking team, and took @danehans 's experience with Contour operator into account when we decided to go forward with kubebuilder + controller-runtime, as opposed to Operator SDK. Kubebuilder + controller-runtime seems to be the more popular approach upstream.